src/panels/lovelace/strategies/get-strategy.ts

```diff
@@ -0,0 +1,50 @@
+// @ts-nocheck
+import { LovelaceDashboardStrategy } from "../../../data/lovelace";
+
+const CUSTOM_PREFIX = "custom:";
+
+const strategies: { [key: string]: () => Promise<any> } = {
+  default: () => Promise.resolve(SingleViewLovelaceDashboardStrategy),
+  by_view: () => import("./by_view").ByViewLovelaceDashboardStrategy,
+};
+
+export const getLovelaceDashboardStrategy = (
+  name: string
+): Promise<LovelaceDashboardStrategy> => {
+  if (name in strategies) {
+    return strategies[name]();
+  }
+
+  if (!name.startsWith(CUSTOM_PREFIX)) {
+    // This will just generate a view with a single error card
+    return errorStrategy(`Unknown strategy specified: ${name}`);
+  }
+
+  const tag = `ll-strategy-${name.substr(CUSTOM_PREFIX.length)}`;
+
+  return customElements.whenDefined(tag).then(() => customElements.get(tag));
```

Comment:
This is the awesome part of this. We will use the custom elements registry as a dependency manager.